### PR TITLE
adding DebugGrid + grid mixins

### PR DESCRIPTION
**Pull request by namniak**
- adding `?grid` param to the url will enable the `DebugGrid`. It will persist on route change
- added `breakpoint` mixin
- added `columns` functions for each breakpoint

The grid system is controlled from one place `/src/data/grid.json`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,15 @@
+## v5.2.0 (October 18, 2019)
```
Comment (ragu-S) [inline CHANGELOG.md]: We should be using semantic-versioning to handle this, remove v5.2.0 additions in the CHANGELOG.md and rename commit to `feat: adding debug grid + grid mixins`, see guide below

image

`npm run release` will auto update the changelog based on those commit keys (like `feat:, fix:, etc.` and update the tag and the package version

Comment (namniak) [reply CHANGELOG.md]: good point @ragu-S 
I'll update

**Comment by jmckinnell** [comment]
@namniak Where did this land? I know this is ancient history at this point but let's get this in.

**Comment by namniak** [comment]
@jmckinnell this is an old generator. I don't think we should be maintaining both.
in terms of the grid component itself, we need to further brainstorm and agree on a basic approach because there are too many variables that define it:

- is column gap flexible or fixed (fixed px value vs. relative to the base width for device)
- is side padding flexible or fixed (fixed value or a value relative to the base width for device)
- is grid size defined by side padding or content width
- are desktop columns fixed size or stretching like mobile/tablet
- are there multiple desktop breakpoint rules (small, medium, large) involved
- is there a lock size (e.g. 1920px)

basically all these things define a different approach and there's a lot of cleanup needed in both css and js if one of the methods is forced but the project doesn't follow it. 
in short, integration needs re-thinking

**Comment by jmckinnell** [comment]
@namniak Fair points! I'm sad to be putting this repo to pasture. It's nice to have a non-static generator in our back pocket for projects that don't need it. But I can understand the complexity that comes with maintaining both. I'm fine to let this drop.

I'll see what we're doing in the NextJS generator. Even with all of those variables on the grid system, it's good to have a template to work off of.
